### PR TITLE
Support solidity v0.7.x

### PR DESCRIPTION
**Pull request by HenryNguyen5**
## Description

Adds support for solidity 0.7.x in sol-compiler

## Testing instructions

Run `yarn build && yarn test` within the `sol-compiler` directory

## Types of changes

New feature
Adds support for solidity 0.7.x in sol-compiler

## Checklist:

-   [x] Add tests to cover changes as needed.
-   [x] Update documentation as needed.
-   [x] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

**Comment by HenryNguyen5** [comment]
Not too sure why CI is failing here, i built the package locally and linked it against https://github.com/smartcontractkit/chainlink/tree/develop/evm-contracts, which worked fine for compiling 0.4.x -> 0.7.x solidity contracts fine

**Comment by dorothy-zbornak** [comment]
> Not too sure why CI is failing here, i built the package locally and linked it against https://github.com/smartcontractkit/chainlink/tree/develop/evm-contracts, which worked fine for compiling 0.4.x -> 0.7.x solidity contracts fine

Yeah it's just the solc registry timing out. I restarted it. :crossed_fingers:

File: packages/sol-compiler/CHANGELOG.json [hunk 1]
```diff
@@ -1,4 +1,13 @@
 [
+    {
+        "timestamp": 1600807895,
```
Comment (dorothy-zbornak) [inline packages/sol-compiler/CHANGELOG.json]:
```suggestion
```
This gets auto-added by our publish job.

File: packages/sol-compiler/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "@0x/sol-compiler",
-    "version": "4.1.1",
+    "version": "4.2.0",
```
Comment (dorothy-zbornak) [inline packages/sol-compiler/package.json]:
```suggestion
    "version": "4.1.1",
```
Will be automatically updated by publish job.

**Review by dorothy-zbornak — APPROVED** [review]
Thanks so much!

File: packages/sol-compiler/src/compiler.ts [hunk 3]
```diff
@@ -20,8 +20,12 @@ import * as pluralize from 'pluralize';
 import * as semver from 'semver';
 import { StandardInput } from 'solc';
 import { promisify } from 'util';
-
 import { compilerOptionsSchema } from './schemas/compiler_options_schema';
+import { ContractContentsByPath, ImportPrefixRemappings, SolcWrapper } from './solc_wrapper';
+import { SolcWrapperV04 } from './solc_wrapper_v04';
+import { SolcWrapperV05 } from './solc_wrapper_v05';
+import { SolcWrapperV06 } from './solc_wrapper_v06';
+import { SolcWrapperV07 } from './solc_wrapper_v07';
```
Comment (dorothy-zbornak) [inline packages/sol-compiler/src/compiler.ts]: These need to be moved back down to where they were to appease the linter.

**Comment by HenryNguyen5** [comment]
@dorothy-zbornak should be fixed up now!

**Comment by dorothy-zbornak** [comment]
Gonna merge this into my branch first to fix the annoying linter issues.

**Pull request merged**